### DISCUSSION TITLE
unsuccessful installation of yarn

### DISCUSSION THREAD

**Discussion by davielin2002** [smartcontractkit/full-blockchain-solidity-course-py#1656] [Q&A]
I can't install yarn successfully, so can't ganache. Thus, when I run the code "deploy.py", I can't run that code. Can anybody help me! thanks!!

<img width="714" alt="截屏2022-08-18 11 14 50" src="https://user-images.githubusercontent.com/109844622/185285288-fa40ffb2-e9d4-44d1-8d11-ad2ca61c8181.png">

**Comment by cromewar** [comment]
Hello @davielin2002 
The reason is the npm packages you are installing are not included correctly to your ZSH path.
The easiest way to solve this is reinstalling node using [NVM](https://github.com/nvm-sh/nvm) which is the best way to manage node versions.

1. Uninstall your current nodeJs installation.
2. Install nvm using the instructions on the link above.
3. restart your terminal and try again.

I hope this info might help.

Additional tip, if you are using a M1 Mac, you would like to install homebrew first as well.
